client/src/components/ResponsiveSideBar.js

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import { Drawer, IconButton, Toolbar } from "@material-ui/core";
+import MenuIcon from "@material-ui/icons/Menu";
+import Hidden from "@material-ui/core/Hidden";
+import SearchIcon from "@material-ui/icons/Search";
+import Fab from "@material-ui/core/Fab";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import CloseIcon from "@material-ui/icons/Close";
+import Typography from "@material-ui/core/Typography";
+
+const drawerWidth = 250;
+
+const useStyles = makeStyles((theme) => ({
+    drawer: {
+        [theme.breakpoints.up("sm")]: {
+            width: drawerWidth,
+            flexShrink: 0,
+        },
+    },
+    drawerPaper: {
+        width: drawerWidth,
+    },
+    drawerContainer: {
+        overflow: "auto",
+        padding: theme.spacing(3),
+    },
+    menuButton: {
+        margin: theme.spacing(3),
+        // position: 'absolute',
+    },
+}));
+
+function ResponsiveSideBar(props) {
+    const classes = useStyles();
+    const theme = useTheme();
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    function handleDrawerToggle() {
+        setMobileOpen(!mobileOpen);
```

Comment:
```suggestion
        setMobileOpen(previous => !previous);
```
setState is async, so this is the best practice way of toggling a boolean in state ^

Reply:
That's good to know thanks!! setMobileOpen(!mobileOpen) is actually how Material UI has it in their docs too.